File: java/src/com/nari/algorithm/model/Node.java
package com.nari.algorithm.model;

/**
 * Author: Xu Jiong
 * Date: 2022/5/25 16:44
 * Description:
 */
public class Node {

    private int val;
    private Node prev;
    private Node next;

    public Node(int val) {this.val = val;}

    public int getVal() {
        return val;
    }

    public void setVal(int val) {
        this.val = val;
    }

    public Node getPrev() {
        return prev;
    }

    public void setPrev(Node prev) {
        this.prev = prev;
    }

    public Node getNext() {
        return next;
    }

    public void setNext(Node next) {
        this.next = next;
    }
}

File: scala/src/com/nari/algorithm/model/Node.scala
package com.nari.algorithm
package model

class Node(_value: Int, _left: Node, _right: Node) {

  var value: Int = _value
  var left: Node = _left
  var right: Node = _right

  def this(_value: Int) {
    this(_value, null, null)
  }
}
